workflows/cloudify_system_workflows/snapshots/snapshot_restore.py

```diff
@@ -250,11 +277,19 @@ def _get_snapshot_path(self):
     def _get_existing_plugin_names(self):
         ctx.logger.debug('Collecting existing plugins')
         existing_plugins = self._client.plugins.list(_all_tenants=True)
-        return set((p.archive_name, p['tenant_name'])
-                   for p in existing_plugins)
-
-    def _get_existing_dep_envs(self):
-        return [dep.id for dep in self._client.deployments.list()]
+        return [self._get_plugin_archive_path_id_and_tenant(p)
+                for p in existing_plugins]
+
+    def _get_plugin_archive_path_id_and_tenant(self, plugin):
+        return {
+            'path': os.path.join(
+                '/opt/manager/resources/plugins',
```

Comment:
To add to this: The constants are in cloudify_manager too, so the only way to use them for this is to copy them into mgmtworker (only used in this one place), or to make it part of the mgmtworker env. I feel like neither of those are brilliant... maybe they should be in plugins-common?